NeoML/src/Dnn/DnnOptimization.cpp

```diff
@@ -33,6 +34,17 @@ CDnnOptimizationReport OptimizeDnn( CDnn& dnn, const CDnnOptimizationSettings& s
 	CDnnOptimizationReport report;
 	optimization::CGraph graph( dnn );
 
+	CArray<CBaseLayer*> layers;
+	graph.GetLayers(layers);
+
+	for (CBaseLayer* layer : layers) {
+		CDnnHeadAdapterLayer* head = dynamic_cast<CDnnHeadAdapterLayer*>(layer);
+		if (head != nullptr)
+		{
+			OptimizeDnn(*(head->GetDnnHead()->GetInnerDnn()));
```

Comment:
If there is a only one adapter for this current sub-network, this adapter should be removed and sub-net reconnected directly, as it works for composite in the ` optimization::UnpackComposites`.